server/interfaces/interfaces.go

```diff
@@ -962,26 +962,26 @@ type XcodeLocator interface {
 }
 
 // LRU implements a Least Recently Used cache.
-type LRU interface {
+type LRU[V any] interface {
 	// Inserts a value into the LRU. A boolean is returned that indicates
 	// if the value was successfully added.
-	Add(key, value interface{}) bool
+	Add(key string, value *V) bool
```

Comment:
If we want an LRU holding a primitive type then we have to pass in pointers, vs the plain old primitive values. This feels slightly clunky? Also, strings in particular are already pointers in a sense, so the extra pointer dereference might be inefficient.

wdyt about using `V` instead of `*V` for all of these and then at each call site declaring `LRU[*myStruct]` instead of the current `LRU[myStruct]`? (not sure if there's a best practice here...)

Reply:
I tried that at first.

However there are some tricky part such as `Get(key string) (*V, bool)` where turning `*V -> V` here would make it impossible to return nil value. 🤔 
And we want to avoid `Add()` in one data type and `Get()` in another.

Some popular open-source container types such as https://github.com/zyedidia/generic/blob/master/btree/btree.go would get around it by returning a default value instead of nil.

```go
func (t *Tree[K, V]) search(x *node[K, V], key K, height int) (V, bool) {
	...
	var v V
	return v, false
}
```

I could explore this direction in the next iteration